jobserver/releases.py

```diff
@@ -31,14 +32,26 @@ def handle_release(workspace, backend, backend_user, release_hash, upload):
     try:
         # first, ensure we can extract the zip
         with ZipFile(upload) as archive:
+            try:
+                json.load(archive.open("metadata/manifest.json"))
```

Comment:
Won't this leave the filehandle for the manifest open?

Reply:
No, as it's not actually opening a file - it's a file-like object that reads bytes out of the zipfile, which is the actual open file handle, and handled by the with statement

Reply:
TIL!